galley/pkg/crd/validation/webhook.go

```diff
@@ -441,27 +450,48 @@ func (wh *Webhook) admitPilot(request *admissionv1beta1.AdmissionRequest) *admis
 
 	var obj crd.IstioKind
 	if err := yaml.Unmarshal(request.Object.Raw, &obj); err != nil {
+		scope.Infof("cannot decode configuration: %v", err)
 		reportValidationFailed(request, reasonYamlDecodeError)
 		return toAdmissionResponse(fmt.Errorf("cannot decode configuration: %v", err))
 	}
 
 	schema, exists := wh.descriptor.GetByType(crd.CamelCaseToKebabCase(obj.Kind))
 	if !exists {
+		scope.Infof("unrecognized type %v", obj.Kind)
 		reportValidationFailed(request, reasonUnknownType)
 		return toAdmissionResponse(fmt.Errorf("unrecognized type %v", obj.Kind))
 	}
 
 	out, err := crd.ConvertObject(schema, &obj, wh.domainSuffix)
 	if err != nil {
+		scope.Infof("error decoding configuration: %v", err)
 		reportValidationFailed(request, reasonCRDConversionError)
 		return toAdmissionResponse(fmt.Errorf("error decoding configuration: %v", err))
 	}
 
 	if err := schema.Validate(out.Name, out.Namespace, out.Spec); err != nil {
+		scope.Infof("configuration is invalid: %v", err)
 		reportValidationFailed(request, reasonInvalidConfig)
 		return toAdmissionResponse(fmt.Errorf("configuration is invalid: %v", err))
 	}
 
+	trial := make(map[string]interface{})
+	if err := yaml.Unmarshal(request.Object.Raw, &trial); err != nil {
+		scope.Infof("cannot re-decode configuration: %v", err)
+		reportValidationFailed(request, reasonYamlDecodeError)
+		return toAdmissionResponse(fmt.Errorf("cannot re-decode configuration: %v", err))
+	}
+
+	for key := range trial {
+		if _, ok := validCategories[key]; !ok {
+			scope.Infof("unknown category %q on %s resource %s/%s",
```

Comment:
Is `category` the right term here? From the user's perspective I think this would simply be an incorrectly named field, e.g. `Spec` vs. `spec`.

Reply:
You are right, the Kubernetes term is 'field'.  I will make the change.  https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields